src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -3048,19 +3049,19 @@ public boolean isEmpty()
     public boolean isRowCacheEnabled()
     {
 
-        boolean retval = metadata().params.caching.cacheRows() && CacheService.instance.rowCache.getCapacity() > 0;
+        boolean retval = metadata().params.caching.cacheRows() && DatabaseDescriptor.getRowCacheSizeInMiB() > 0;
```

Comment:
`DatabaseDescriptor.getRowCacheSizeInMiB() > 0` -- this condition is unclear to the user/developer facing it for the first time, so my suggestion is to create a new method in the `DatabaseDescriptor` class and reuse it where we need it. I've looked at the latest branch (trunk) and I think we could push a new method there as well.

An example:
```
    public static boolean isRowCacheEnabled()
    {
        return conf.row_cache_size.toMebibytes() > 0;
    }
```

Similar methods should be created for the counter and key caches.

Reply:
I think for the row cache the correct condition is the following:
`getRowCacheClassName() =! null && conf.row_cache_size.toMebibytes() > 0`

As the row cache is pluggable we should also check the pluggable class is not null.